tests/test_worker.py

```diff
@@ -621,3 +625,33 @@ def test_working_worker_cold_shutdown(self):
         shutdown_requested_date = w.shutdown_requested_date
         self.assertIsNotNone(shutdown_requested_date)
         self.assertEqual(type(shutdown_requested_date).__name__, 'datetime')
+
+    @slow
+    def test_work_horse_death_sets_job_failed(self):
+        """worker with an ongoing job whose work horse dies unexpectadly should
+        set the job's status either to FINISHED or FAILED
+        """
+        fooq = Queue('foo')
+        failed_q = get_failed_queue()
+        self.assertEqual(failed_q.count, 0)
+        self.assertEqual(fooq.count, 0)
+        w = Worker(fooq)
+        registry = StartedJobRegistry(connection=self.testconn)
+        sentinel_file = '/tmp/.rq_sentinel_work_horse_death'
+        if os.path.exists(sentinel_file):
+            os.remove(sentinel_file)
+        fooq.enqueue(create_file_after_timeout, sentinel_file, 100)
+        job, queue = w.dequeue_job_and_maintain_ttl(5)
+        w.fork_work_horse(job, queue)
+        p = Process(target=kill_work_horse, args=(w._horse_pid,))
+        p.start()
+        p.join(1)
+        w.monitor_work_horse(job)
```

Comment:
I know this is very difficult to test, but this doesn't test the case where the process terminates unexpectedly after `w.monitor_work_horse(job)` is called.